R/ProjectConfiguration.R

```diff
@@ -0,0 +1,78 @@
+#' @title ProjectConfiguration
+#' @docType class
+#' @description An object storing configuration used project-wide
+#' @format NULL
+ProjectConfiguration <- R6::R6Class(
+  "ProjectConfiguration",
+  inherit = ospsuite.utils::Printable,
+  cloneable = TRUE,
+  active = list(),
+  private = list(),
+  public = list(
+    #' @description
+    #' Initialize a new instance of the class
+    #' @return A new `ProjectConfiguration` object.
+    initialize = function() {
+    },
+
+    #' @field modelFolder Path to the folder with pkml simulation files;
+    #' relative to the "Code" folder.
+    modelFolder = NULL,
+    #' @field paramsFolder Path to the folder with excel files with
+    #' parametrization; relative to the "Code" folder.
+    paramsFolder = NULL,
+    #' @field paramsFile Name of the excel file with global model
+    #' parametrization. Must be located in the "paramsFolder".
+    paramsFile = NULL,
+    #' @field individualParamsFile Name of the excel file with
+    #' individual-specific model parametrization. Must be located in the
+    #' "paramsFolder"
+    individualParamsFile = NULL,
+    #' @field individualPhysiologyFile Name of the excel file with
+    #' individual physiology information. Must be located in the "paramsFolder".
+    individualPhysiologyFile = NULL,
+    #' @field populationParamsFile Name of the excel file with population
+    #' information. Must be located in the "paramsFolder".
+    populationParamsFile = NULL,
+    #' @field scenarioDefinitionFile Name of the excel file with scenario
+    #' definitions. Must be located in the "paramsFolder".
+    scenarioDefinitionFile = NULL,
+    #' @field scenarioApplicationsFile Name of the excel file scenario-specific
+    #'  parameters such as application protocol parameters. Must be located in
+    #'   the "paramsFolder".
+    scenarioApplicationsFile = NULL,
+    #' @field dataFolder Path to the folder where experimental data files are located
+    dataFolder = NULL,
+    #' @field dataFile Name of the excel file with experimental data
+    dataFile = NULL,
+    #' @field dataImporterConfigurationFile Name of data importer configuration
+    #' file in xml format used to load the data. Must be located in the "dataFolder"
+    dataImporterConfigurationFile = NULL,
+    #' @field compoundPropertiesFile Path to the excel file containing information
+    #' about all compounds in the model. Must be located in the "dataFolder"
+    compoundPropertiesFile = NULL,
+    #' @field outputFolder Path to the folder where the results should be
+    #' saved to; relative to the "Code" folder
```

Comment:
I think this can be moved to the top so that all `"Code"` folder related public fields are together; the same way that all fields related to `"paramsFolder"` are contiguous.

Reply:
The idea was to group the fields by the folders and their contents. So first the folder "modelFolder", then the folder "paramsFolder" following all files located in this folder, then "dataFolder" and the files within, and finally the "outputFolder"